From: Salvatore Bonaccorso <carnil@debian.org>
Date: Fri, 22 Nov 2024 20:40:39 +0100
Subject: Revert "x86: Increase brk randomness entropy for 64-bit systems"
Origin: https://lore.kernel.org/stable/202411210628.ECF1B494D7@keescook/
Bug-Debian: https://bugs.debian.org/1085762

This reverts commit b0cde867b80a5e81fcbc0383e138f5845f2005ee which is
commit 44c76825d6eefee9eb7ce06c38e1a6632ac7eb7d upstream.

There seems to be a bad interaction between this change and older
PIE-built qemu-user-static (for aarch64) binaries.

Link: https://lore.kernel.org/all/202411201000.F3313C02@keescook/
Link: https://bugs.debian.org/1085762
Signed-off-by: Salvatore Bonaccorso <carnil@debian.org>
---
 arch/x86/kernel/process.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/arch/x86/kernel/process.c b/arch/x86/kernel/process.c
index acc83738bf5b..279b5e9be80f 100644
--- a/arch/x86/kernel/process.c
+++ b/arch/x86/kernel/process.c
@@ -991,10 +991,7 @@ unsigned long arch_align_stack(unsigned long sp)
 
 unsigned long arch_randomize_brk(struct mm_struct *mm)
 {
-	if (mmap_is_ia32())
-		return randomize_page(mm->brk, SZ_32M);
-
-	return randomize_page(mm->brk, SZ_1G);
+	return randomize_page(mm->brk, 0x02000000);
 }
 
 /*
-- 
2.45.2

